pkg/config/validation/validation.go

```diff
@@ -1209,7 +1209,7 @@ var ValidateSidecar = registerValidateFunc("ValidateSidecar",
 					}
 					udsMap[bind] = struct{}{}
 				} else {
-					if _, found := portMap[egress.Port.Number]; found {
+					if _, found := portMap[egress.Port.Number]; found || len(egress.Hosts) > 1 {
```

Comment:
Perhaps I need to validate against protocol here too. For the TCP case, the resulting listener maps directly to an upstream cluster and if there are multiple hosts only one of the clusters can be used - I guess HTTP and other protocols are different in that the listeners map to routes rather than clusters

Reply:
I think tcp service can also use same port. We generate listener with VIP:port address

Reply:
it seems the listener is only generated with VIP:port if there is no port specified on the EgressListener - for an EgressListener with multiple hosts and one port the actual listener will be 0.0.0.0:port -> <cluster>. If the port isn't specified, and the multiple hosts are moved to the catch all block, then the listeners are correctly generated as VIP:port for each host